iidm/iidm-modification/src/test/java/com/powsybl/iidm/modification/Replace3TwoWindingsTransformersByThreeWindingsTransformersTest.java

```diff
@@ -0,0 +1,434 @@
+
+        TwtData t3wData = new TwtData(t3w, 0.0, false);
+        setStarBusVoltage(t3wData, t2w1.getTerminal2().getBusView().getBus());
+
+        BranchData t2w1Data = new BranchData(t2w1, 0.0, false, false);
+        BranchData t2w2Data = new BranchData(t2w2, 0.0, false, false);
+        BranchData t2w3Data = new BranchData(t2w3, 0.0, false, false);
+
+        double tol = 0.000001;
+        assertEquals(t2w1Data.getComputedP1(), t3wData.getComputedP(ThreeSides.ONE), tol);
+        assertEquals(t2w2Data.getComputedP2(), t3wData.getComputedP(ThreeSides.TWO), tol);
+        assertEquals(t2w3Data.getComputedP1(), t3wData.getComputedP(ThreeSides.THREE), tol);
+        return true;
+    }
+
+    @Test
+    void replaceNodeBreakerTest() {
+        modifyNetworkForNodeBreakerTest();
+
+        assertEquals(4, network.getVoltageLevelCount());
+        assertEquals(3, network.getTwoWindingsTransformerCount());
+        assertEquals(0, network.getThreeWindingsTransformerCount());
+
+        Replace3TwoWindingsTransformersByThreeWindingsTransformers replace = new Replace3TwoWindingsTransformersByThreeWindingsTransformers();
+        replace.apply(network);
+
+        assertEquals(3, network.getVoltageLevelCount());
+        assertEquals(0, network.getTwoWindingsTransformerCount());
+        assertEquals(1, network.getThreeWindingsTransformerCount());
+    }
+
+    private void modifyNetworkForNodeBreakerTest() {
+        Network expectedNetwork = createThreeWindingsTransformerNodeBreakerNetwork();
+        ReplaceThreeWindingsTransformersBy3TwoWindingsTransformers replaceExpected = new ReplaceThreeWindingsTransformersBy3TwoWindingsTransformers();
+        replaceExpected.apply(expectedNetwork);
+        t2w1 = expectedNetwork.getTwoWindingsTransformer("3WT-Leg1");
+        assertNotNull(t2w1);
+        t2w2 = expectedNetwork.getTwoWindingsTransformer("3WT-Leg2");
+        assertNotNull(t2w2);
+        t2w3 = expectedNetwork.getTwoWindingsTransformer("3WT-Leg3");
+
+        network = createThreeWindingsTransformerNodeBreakerNetwork();
+        ReplaceThreeWindingsTransformersBy3TwoWindingsTransformers replace = new ReplaceThreeWindingsTransformersBy3TwoWindingsTransformers();
+        replace.apply(network);
+    }
+
+    @Test
+    void applyChecks2Test() {
+        network.getTwoWindingsTransformer(t2w1.getId()).setProperty("unknown property", "unknown value");
+        network.getTwoWindingsTransformer(t2w2.getId()).setProperty("unknown property", "unknown value");
+        Replace3TwoWindingsTransformersByThreeWindingsTransformers replace = new Replace3TwoWindingsTransformersByThreeWindingsTransformers();
+        assertThrows(PowsyblException.class, () -> replace.apply(network, true, ReportNode.NO_OP),
+                "An unknown property should fail to apply.");
+        assertDoesNotThrow(() -> replace.apply(network, false, ReportNode.NO_OP),
+                "An unknown property should not throw if throwException is false.");
```

Comment:
If we keep the `logOrThrow` for non-transferred properties (see previous comment), the 2 error messages must be changed since they do no more reflect the encountered problem.

(Same applies for the `applyChecks3Test` method.)

Reply:
Removed as only a warning was emitted.